client/lib/use-in-view/index.ts

```diff
@@ -1,21 +1,20 @@
 import { RefObject, useEffect, useRef } from 'react';
 
 export function useInView< T extends Element >( oneTimeCallback: () => void ): RefObject< T > {
-	const viewedRef = useRef< boolean >( false );
+	const viewedRef = useRef( false );
 	const elementRef = useRef< T >( null );
-	const observerRef = useRef< IntersectionObserver >();
+	const oneTimeCallbackRef = useRef< () => void >();
+
+	useEffect( () => {
+		oneTimeCallbackRef.current = oneTimeCallback;
+	}, [ oneTimeCallback ] );
 
 	useEffect( () => {
 		// We can't do anything without a valid reference to an element on the page
 		if ( ! elementRef.current ) {
 			return;
 		}
 
-		// If the observer is already defined, no need to continue
-		if ( observerRef.current ) {
-			return;
-		}
-
 		const handler = ( entries: IntersectionObserverEntry[] ) => {
 			// Only fire once per page load
 			if ( viewedRef.current ) {
```

Comment:
Can we get rid of this `viewedRef`? We want to fire the callback only once and it's triggered by the intersection observer, so isn't it a matter of disconnecting the observer once the callback is called, as we're doing that already by calling `observer.disconnect()`?

In fact, we don't need to declare the refs as dependencies of the `useEffect` array since the ref wrappers itself are stable -- the linter doesn't complain 😄 So we won't see the effect running multiple times and disconnecting it will prevent it from running several times.

Reply:
Fixed.